### PR TITLE
SCP-4726 Migrate to nix flakes

### PR DESCRIPTION
**Pull request by hrajchert**
This PR includes changes from [PR-19](https://github.com/input-output-hk/marlowe-playground/pull/19) and unblocks [PR-16](https://github.com/input-output-hk/marlowe-playground/pull/16), which could be reviewed at the same time.

It migrates the nix code to flakes, removing the update materialized and updating the marlowe-cardano dependency. 

Deployment configuration is not included in this PR.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [X] Commit sequence broadly makes sense
    - [X] Key commits have useful messages
    - [X] Relevant tickets are mentioned in commit messages
    - [X] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [X] Self-reviewed the diff
    - [X] Useful pull request description
    - [X] Reviewer requested

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,95 +1,181 @@
...
+                  {
+                    packages.plutus-script-utils.ghcOptions = [ "-Wwarn" "-Wno-unused-packages" ];
+                    # See https://github.com/input-output-hk/iohk-nix/pull/488
+                    packages.cardano-crypto-praos.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
+                    packages.cardano-crypto-class.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
+                  }
+                ];
+                shell.tools = {
+                  cabal = { };
+                  # hlint = {};
```
Comment (jhbertra) [inline flake.nix]: How come this is commented out?

File: marlowe-playground-server/marlowe-playground-server.cabal [hunk 2]
```diff
@@ -107,31 +121,28 @@ executable marlowe-playground-server
...
-        time -any,
         time-units -any
+        -- marlowe-actus -any,
+        -- plutus-tx,
+        -- servant-foreign -any,
+        -- servant-client -any,
+        -- http-client -any,
+        -- http-types -any,
+        -- wai -any,
+        -- time -any,
```
Comment (jhbertra) [inline marlowe-playground-server/marlowe-playground-server.cabal]: Why leave these here?

File: marlowe-playground-server/marlowe-playground-server.cabal [hunk 3]
```diff
@@ -54,32 +67,33 @@ library
...
-        validation -any
+        unordered-containers -any,
+        wai-cors -any
+        -- directory -any,
+        -- marlowe-symbolic -any,
+        -- marlowe-actus -any,
+        -- servant-client-core -any,
+        -- stm -any,
+        -- uuid -any,
+        -- validation -any
```
Comment (jhbertra) [inline marlowe-playground-server/marlowe-playground-server.cabal]: Why leave these here? If they aren't needed, they don't need to be commented out.

File: nix/fix-stylish-haskell/default.nix [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+writeShellScriptBin "fix-stylish-haskell" ''
+  # stylish-haskell can fail for some files that it can't parse, and fd will terminate
+  # if any of the exec commands fail. Rather than blacklisting the bad files, we simply
+  # ignore the exit code, which is usually unhelpful.
+  ${fd}/bin/fd \
+    --extension hs \
+    --exclude 'dist-newstyle/*' \
+    --exclude 'dist/*' \
+    --exclude '.stack-work/*' \
+    --exec bash -c "${stylish-haskell}/bin/stylish-haskell -i {} || true"
```
Comment (jhbertra) [inline nix/fix-stylish-haskell/default.nix:12]: Looking at this again, this is not a good way to handle this! I would rather this test fails if `stylish-haskell` fails. If it fails to parse a module, it means that a Language extension is enabled by default in the cabal project that is not listed in the stylish-haskell config, which is a problem to be addressed, not ignored.

**Review by jhbertra — APPROVED** [review]
Overall, looks great! It would be nice to get materialization working though, if you want an example of getting this working with a flake setup, check out this project https://github.com/shlevy/eventuo11y/blob/master/flake.nix#L59

File: flake.nix (hunk 1 above)
Comment (hrajchert) [reply flake.nix]: We are not using it (for the moment)... removing it

File: marlowe-playground-server/marlowe-playground-server.cabal (hunk 2 above)
Comment (hrajchert) [reply marlowe-playground-server/marlowe-playground-server.cabal]: My bad, I commented all of them and started uncommenting the necesary ones, but forgot to delete the old ones.

File: marlowe-playground-server/marlowe-playground-server.cabal (hunk 3 above)
Comment (hrajchert) [reply marlowe-playground-server/marlowe-playground-server.cabal]: same

File: nix/fix-stylish-haskell/default.nix (hunk 4 above)
Comment (hrajchert) [reply nix/fix-stylish-haskell/default.nix:12]: I just copied the definition in plutus app that we were already using (I copied to remove the nix coupling).
I'm creating ticket SCP-4786 to address this in a future PR.

**Pull request merged**